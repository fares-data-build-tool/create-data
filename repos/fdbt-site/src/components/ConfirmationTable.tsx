import React, { ReactElement } from 'react';

interface ConfirmationTableProps {
    confirmationElements: ConfirmationElement[];
    header: string;
}

export interface ConfirmationElement {
    name: string;
    content: string;
    href: string;
}

const ConfirmationTable = ({ confirmationElements, header }: ConfirmationTableProps): ReactElement => {
    const builtElements = confirmationElements.map(element => (
        <React.Fragment key={header}>
            <dl className="govuk-summary-list">
                <div className="govuk-summary-list__row" key={element.name}>
                    <dt className="govuk-summary-list__key">{element.name}</dt>
                    <dd className="govuk-summary-list__value">{element.content}</dd>
                    <dd className="govuk-summary-list__actions">
                        {element.href !== '' ? (
                            <a className="govuk-link" href={element.href}>
                                change<span className="govuk-visually-hidden">{element.name}</span>
                            </a>
                        ) : null}
                    </dd>
                </div>
            </dl>
        </React.Fragment>
    ));

    return (
        <>
            <div className="govuk-warning-text">
                <span className="govuk-warning-text__icon" aria-hidden="true">
                    !
                </span>
                <strong className="govuk-warning-text__text">
                    <span className="govuk-warning-text__assistive">Warning</span>
                    You will not be shown these answers again, ensure they are correct
                </strong>
            </div>
            {builtElements}
        </>
    );
};

export default ConfirmationTable;
