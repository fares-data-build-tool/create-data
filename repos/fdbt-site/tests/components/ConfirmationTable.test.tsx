import { shallow } from 'enzyme';
import React from 'react';
import ConfirmationTable from '../../src/components/ConfirmationTable';

describe('ConfirmationTable', () => {
    it('should render the table with options', () => {
        const wrapper = shallow(
            <ConfirmationTable
                confirmationElements={[
                    {
                        header: 'Snapshot test things',
                        innerElements: [
                            { name: 'Name', content: 'Bob', href: '/login' },
                            { name: 'House', content: 'Big', href: '/fareType' },
                        ],
                    },
                    {
                        header: 'Other Snapshot test things',
                        innerElements: [{ name: 'Age', content: '21', href: '/' }],
                    },
                ]}
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });
});
