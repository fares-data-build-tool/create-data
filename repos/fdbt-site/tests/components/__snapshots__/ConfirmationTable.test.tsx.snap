// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationTable should render the table with options 1`] = `
<Fragment>
  <div
    className="govuk-warning-text"
  >
    <span
      aria-hidden="true"
      className="govuk-warning-text__icon"
    >
      !
    </span>
    <strong
      className="govuk-warning-text__text"
    >
      <span
        className="govuk-warning-text__assistive"
      >
        Warning
      </span>
      You will not be shown these answers again, ensure they are correct
    </strong>
  </div>
  <dl
    className="govuk-summary-list"
  >
    <div
      className="govuk-summary-list__row"
      key="Name"
    >
      <dt
        className="govuk-summary-list__key"
      >
        Name
      </dt>
      <dd
        className="govuk-summary-list__value"
      >
        Bob
      </dd>
      <dd
        className="govuk-summary-list__actions"
      >
        <a
          className="govuk-link"
          href="/login"
        >
          change
          <span
            className="govuk-visually-hidden"
          >
            Name
          </span>
        </a>
      </dd>
    </div>
  </dl>
  <dl
    className="govuk-summary-list"
  >
    <div
      className="govuk-summary-list__row"
      key="House"
    >
      <dt
        className="govuk-summary-list__key"
      >
        House
      </dt>
      <dd
        className="govuk-summary-list__value"
      >
        Big
      </dd>
      <dd
        className="govuk-summary-list__actions"
      >
        <a
          className="govuk-link"
          href="/fareType"
        >
          change
          <span
            className="govuk-visually-hidden"
          >
            House
          </span>
        </a>
      </dd>
    </div>
  </dl>
</Fragment>
`;
